File: CSVParser.py
#!/usr/local/bin/python
import re
def scrub_quote(d):
    d = re.sub(r"\"","", d)
    d = re.sub(r"\\r\\n","",d)
    d = re.sub(r"\\n","",d)
    d = "\"" + d + "\""
    return d

file = open("testdata.txt", "r")
data = file.readlines()
file.close()
count=0
header=[]
records=[]

for i in data:
    if (count == 0):
         header = i.split(",")
         next
    else:
        rec = i.split(",")
        m_obj = {}
        index = 0
        for i in header:
            m_obj[i] =rec[index]
            index+=1
        records.append(m_obj)
    count+=1
    #print count
final = []
for i in records:
    fmt=[]
    s=""
    for k in i.keys():
        s = ("%s : %s" % (scrub_quote(k.strip()), scrub_quote(i[k].strip())))
        fmt.append(s)
    recStr= "{" +  ",".join(fmt) + "}"
    final.append(recStr)
print '{\n  "payload":[\n   ' +  ',\n'.join(final) + '\n  ]\n}\n'

File: CSVParser.pl
#!/usr/bin/perl
open(F,"testdata.txt") || die $!;
@lines = <F>;
close F;
chomp @lines;
#print @lines;
my @records;
@header = split /,/, @lines[0];
map {s/^\s+//;} @header;
for (my $r=1; $r<$#lines;$r++) {
    @data = split /,/, @lines[$r];
    map {
        s/"(.*)"/$1/;
        s/^[\s+]?(.*)[\s+]?$/$1/;
        s/\\r\\n//;
        s/\\n//;
    } @data;
    my $rec={};
    for ( my $obj=0; $obj<=$#header;$obj++) {
      $rec->{$header[$obj]} = $data[$obj];
    }
    push @records, $rec;
}
my @formattedRecords;
foreach my $rec (@records) {
    my $obj = "{";
    foreach my $k (keys $rec) {
        $obj.= "\"".$k."\":\"".$rec->{$k}."\",";
    }
    $obj=~ s/,$//;
    $obj.="}\n";
    push @formattedRecords, $obj;
}
print "{\n  \"payload\":[";
print join (",", @formattedRecords);
print "  ]\n}\n";
